File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;

        for(char x: s)
            if(x == ')' || x == '}' || x == ']')
                if(st.empty()) return false;
                else if(x == ')' && st.top() != '(') return false;
                else if(x == '}' && st.top() != '{') return false;
                else if(x == ']' && st.top() != '[') return false;
                else st.pop();
            else st.push(x);
        return st.empty() ? true : false;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c == '}' or c == ']' or c == ')':
                if len(stack) == 0:
                    return False
                elif stack[-1] == '(' and c != ')':
                    return False
                elif stack[-1] == '{' and c != '}':
                    return False
                elif stack[-1] == '[' and c != ']':
                    return False
                else:
                    stack.pop()
            else:
                stack.append(c)
        if len(stack) == 0:
            return True
        else:
            return False
                
                